implementations/rust/node/src/node.rs

```diff
@@ -115,78 +141,161 @@ pub fn get_route() -> Option<Route> {
+        if let Ok(sa) = SocketAddr::from_str(&rs) {
+            if let Some(ra) = RouterAddress::from_address(Address::UdpAddress(sa)) {
+                route.addresses.push(ra);
+            }
+        }
     };
-    let command = OckamCommand::Channel(ChannelCommand::SendMessage(m));
-    match channel_tx_for_node.send(command) {
-        Ok(_unused) => {}
-        Err(_unused) => {}
+
+    if let Some(wa) = args.worker_addr {
+        if let Ok(ra) = RouterAddress::worker_router_address_from_str(&wa) {
+            route.addresses.push(ra);
+        }
+    };
+
+    let mut message = "Hello Ockam".to_string();
+    if let Some(m) = args.message {
+        message = m;
+    };
+
+    Ok((local_socket, route, message))
+}
+
+pub fn start_node(local_socket: RouterAddress, onward_route: Route, payload: String) {
+    let (transport_tx, transport_rx) = std::sync::mpsc::channel();
+    let (router_tx, router_rx) = std::sync::mpsc::channel();
+    let (worker_tx, worker_rx) = std::sync::mpsc::channel();
+    let worker_tx_for_node = worker_tx.clone();
+    let router_tx_for_worker = router_tx.clone();
+
+    let mut router = Router::new(router_rx);
+    let mut worker = TestWorker::new(worker_rx, worker_tx, router_tx_for_worker);
+
+    let sock_str: String;
+    match local_socket.address {
+        Address::UdpAddress(udp) => {
+            sock_str = udp.to_string();
+            println!("{}", udp.to_string());
+        }
+        _ => return,
     }
 
-    sleep(time::Duration::from_millis(5000));
+    let mut transport =
+        UdpTransport::new(transport_rx, transport_tx, router_tx, &sock_str).unwrap();
+
+    let _join_thread: thread::JoinHandle<_> = thread::spawn(move || {
+        while transport.poll() && router.poll() & worker.poll() {
+            thread::sleep(time::Duration::from_millis(100));
+        }
+    });
 
-    let command = TransportCommand::Stop;
-    match transport_tx_for_node.send(OckamCommand::Transport(command)) {
-        Ok(_unused) => println!("sent stop command to transport"),
-        Err(_unused) => println!("failed to send command to transport"),
+    if !onward_route.addresses.is_empty() && !payload.is_empty() {
+        let m = Message {
+            onward_route,
+            return_route: Route { addresses: vec![] },
+            message_type: MessageType::Payload,
+            message_body: payload.as_bytes().to_vec(),
+        };
+        let command = OckamCommand::Worker(WorkerCommand::SendMessage(m));
+        match worker_tx_for_node.send(command) {
+            Ok(_unused) => {}
+            Err(_unused) => {
+                println!("failed send to channel");
+            }
+        }
     }
+}
 
-    match join_thread.join() {
-        Ok(_unused) => {}
-        Err(_unused) => {}
+fn main() {
+    let args = Args::from_args();
+    println!("{:?}", args);
+    let local_socket: RouterAddress;
+    let route: Route;
+    let message: String;
+    match parse_args(args) {
```

Comment:
Not that important, but you could remove all the let bindings, and pass the tuple values from `Ok` match arm into the `start_node` call and sleep from there. It's common in Rust to try to avoid doing a declaration without assignment. Even though the compiler will reject it, this could ultimately lead to using uninitialized vars.